src/Augurk/Controllers/V1/FeatureController.cs

```diff
@@ -53,57 +51,37 @@ public async Task<DisplayableFeature> GetAsync(string branchName, string groupNa
 
         [Route("api/features/{branchName}/{groupName}/{title}")]
         [HttpPost]
-        public async Task<HttpResponseMessage> PostAsync(Feature feature, string branchName, string groupName, string title)
+        public async Task<ActionResult<Feature>> PostAsync(Feature feature, string branchName, string groupName, string title)
         {
             if (!feature.Title.Equals(title, StringComparison.OrdinalIgnoreCase))
             {
-                return Request.CreateResponse(HttpStatusCode.BadRequest, "The title provided by the POST data and the title in uri do not match!");
-            }
-
-            var response = Request.CreateResponse(HttpStatusCode.Created);
-
-            try
-            {
-                // NOTE: Using the branchName as product name because of backwards compatability
-                await _featureManager.InsertOrUpdateFeatureAsync(feature, branchName, groupName, UNKNOWN_VERSION);
-            }
-            catch (Exception exception)
-            {
-                response = Request.CreateErrorResponse(HttpStatusCode.InternalServerError, exception);
```

Comment:
I've removed these try/catches since I think that throwing an exception will result in an InternalServerError anyway, so why handle these differently?

Reply:
Fair enough. There was a time in the early days of WebAPI where this was considered "clean".